tests/modules/mcuboot/key-locking/src/main.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/ztest.h>
+#include <stdio.h>
+
+#include <psa/crypto.h>
+#include <psa/crypto_types.h>
+#include <cracen_psa_kmu.h>
+#include <zephyr/sys/util.h>
+
+#define KEY_LIFETIME	PSA_KEY_LIFETIME_FROM_PERSISTENCE_AND_LOCATION(PSA_KEY_PERSISTENCE_DEFAULT, PSA_KEY_LOCATION_CRACEN_KMU)
+#define MAKE_PSA_KMU_KEY_ID(id) PSA_KEY_HANDLE_FROM_CRACEN_KMU_SLOT(CRACEN_KMU_KEY_USAGE_SCHEME_RAW, id)
+
+/* Testing only the first key slot dedicated to ED25519 */
+#define DEFAULT_TEST_KEY MAKE_PSA_KMU_KEY_ID(226)
+
+/* Public key here is generated from bootloader/mcuboot/root-ed25519.pem
+ * identified by SHA256
+ * f1c516ce7e6ed759e08ede0d81b26ae0074b1ee1eb2fd083e545acd7b0cc3eea.
+ * using command:
+ *  openssl pkey -in bootloader/mcuboot/root-ed25519.pem -pubout -outform DER | \
+ *    tail -c 32 | hexdump --format '8/1 "0x%02x, " "\n"'
+ *
+ */
+static const uint8_t ed25519_key[] = {
+	0xd4, 0xb3, 0x1b, 0xa4, 0x9a, 0x3a, 0xdd, 0x3f,
+	0x82, 0x5d, 0x10, 0xca, 0x7f, 0x31, 0xb5, 0x0b,
+	0x0d, 0xe8, 0x7f, 0x37, 0xcc, 0xc4, 0x9f, 0x1a,
+	0x40, 0x3a, 0x5c, 0x13, 0x20, 0xff, 0xb4, 0xe0,
+};
+
+/* Message for signature verification; generated using command:
+ *  echo "Hello World PSA!" | hexdump --format '8/1 "0x%02x, " "\n"' | \
+ *    sed -e 's/0x  ,//g'
+ */
+static const uint8_t message[] = {
+	0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x57, 0x6f,
+	0x72, 0x6c, 0x64, 0x20, 0x50, 0x53, 0x41, 0x21,
+	0x0a,
+};
+
+/* Signature of above message, generated using command:
+ *  MSG=`mktemp`; echo "Hello World PSA!" > $MSG; \
+ *    openssl pkeyutl -sign -inkey bootloader/mcuboot/root-ed25519.pem \
+ *    -rawin -in $MSG | hexdump --format '8/1 "0x%02x, " "\n"'
+ */
+static const uint8_t signature[] = {
+	0xce, 0xf2, 0xbd, 0xb8, 0xdb, 0x68, 0xf8, 0xd8,
+	0x14, 0x3b, 0x63, 0x6e, 0x69, 0x27, 0x02, 0xc4,
+	0x02, 0xb4, 0x99, 0x5e, 0xf8, 0x43, 0x6b, 0x9f,
+	0xf1, 0x8b, 0x72, 0x77, 0x7f, 0x04, 0xfe, 0xb3,
+	0x39, 0x1b, 0x6e, 0x7b, 0x73, 0xad, 0xb6, 0xe2,
+	0x87, 0x23, 0x9e, 0xb9, 0xfa, 0x20, 0x67, 0x40,
+	0xa4, 0xd6, 0x3a, 0xf6, 0x0d, 0x5e, 0xfa, 0x83,
+	0xa6, 0x43, 0xc0, 0x83, 0xff, 0x4b, 0xbd, 0x0a,
+};
+
+static void *test_init(void)
+{
+	zassert_equal(PSA_SUCCESS, psa_crypto_init());
+
+	return NULL;
+}
+
+ZTEST(key_lock_test, test_psa_destroy)
+{
+	psa_status_t status;
+
+	status = psa_destroy_key(DEFAULT_TEST_KEY);
```

Comment:
here `psa_destroy_key` is used. This would constitute a revocation, not a lock...

Shouldn't this use `cracen_kmu_block` ?